runtime/fail_nat.c

```diff
@@ -49,35 +48,37 @@ extern caml_generated_constant
   caml_exn_Sys_blocked_io,
   caml_exn_Stack_overflow,
   caml_exn_Assert_failure,
-  caml_exn_Undefined_recursive_module;
+  caml_exn_Undefined_recursive_module,
+  caml_exn_Unhandled,
+  caml_exn_Continuation_already_taken;
 
 /* Exception raising */
 
-CAMLnoreturn_start
-  extern void caml_raise_exception (caml_domain_state* state, value bucket)
-CAMLnoreturn_end;
+extern void caml_raise_exception (caml_domain_state* state, value bucket)
+  Noreturn;
```

Comment:
Please use the trunk version of this code.  `Noreturn` should no longer be used, as it cannot be implemented properly in MSVC.
```suggestion
CAMLnoreturn_start
  extern void caml_raise_exception (caml_domain_state* state, value bucket)
CAMLnoreturn_end;
```